### PR TITLE
helm3 API in-process

### PR DESCRIPTION
**Pull request by andriy-bulynko**

### CODE REVIEW

**Review by ashwathishiva — APPROVED** [review]

File: api/builtins_qlik/HelmChart.go [hunk 1]
```diff
@@ -27,349 +39,385 @@ type HelmChartPlugin struct {
...
 			return nil, err
 		}
 	}
 
 	return p.rf.NewResMapFromBytes(templatedYaml)
 }
 
-func (p *HelmChartPlugin) deleteRequirements(dir string) error {
+func (p *HelmChartPlugin) executeHelmTemplate() ([]byte, error) {
+	const repoName = "qlik"
```
Comment (bearium) [inline api/builtins_qlik/HelmChart.go]: Should this really be a const? would it not be better to pass in the config?

File: api/builtins_qlik/HelmChart.go [hunk 2]
```diff
@@ -27,349 +39,385 @@ type HelmChartPlugin struct {
...
+		p.logger.Printf("error updating helm repos, err: %v\n", err)
+		return nil, err
+	}
+
+	if err := p.helmFetchIfRequired(settings, repoName); err != nil {
+		p.logger.Printf("error fetching from helm, err: %v\n", err)
+		return nil, err
+	}
+
+	subChartPath := filepath.Join(p.ChartHome, p.ChartName, "charts", p.SubChart)
```
Comment (bearium) [inline api/builtins_qlik/HelmChart.go]: are we assuming all charts have sub-charts?

File: api/builtins_qlik/HelmChart.go (hunk 1 above)
Comment (andriy-bulynko) [reply api/builtins_qlik/HelmChart.go]: Good point! Will address.

File: api/builtins_qlik/HelmChart.go (hunk 2 above)
Comment (andriy-bulynko) [reply api/builtins_qlik/HelmChart.go]: Good point too! Addressing.

**Review by bearium — APPROVED** [review]

**Comment by bearium** [comment]
LGTM ran tests on my side and they worked as expected.

File: api/builtins_qlik/HelmChart.go (hunk 1 above)
Comment (andriy-bulynko) [reply api/builtins_qlik/HelmChart.go]: @bearium: Please take a look at the updated code when you get a chance.

File: api/builtins_qlik/HelmChart.go (hunk 2 above)
Comment (andriy-bulynko) [reply api/builtins_qlik/HelmChart.go]: @bearium: Please take a look at the updated code when you get a chance.

File: api/builtins_qlik/HelmChart.go (hunk 1 above)
Comment (bearium) [reply api/builtins_qlik/HelmChart.go]: @andriy-bulynko new stuff looks and works as expected

**Pull request merged**